### PR TITLE
Feature/781 breadcrumbs

### PR DESCRIPTION
**Pull request by Diaan**

### CODE REVIEW

File: packages/components/breadcrumbs/src/breadcrumbs.scss [hunk 1]
```diff
@@ -29,10 +30,26 @@ a,
...
+a:hover,
+::slotted(a:hover) {
+  --_link-color: var(--sl-color-href-hover-foreground);
+}
+
+a:active,
+::slotted(a:active) {
+  --_link-color: var(--sl-color-href-active-foreground);
+}
+
```
Comment (anna-lach) [inline packages/components/breadcrumbs/src/breadcrumbs.scss:52]: It looks like the outline with tokens is missing in focus-visible?

image

File: packages/components/breadcrumbs/src/breadcrumbs.stories.ts [hunk 2]
```diff
@@ -29,7 +31,7 @@ export const Basic: Story = {
     breadcrumbs: html`
       <a href="javascript:void(0)">Lorem</a>
       <a href="javascript:void(0)">Ipsum</a>
-      <span>Dolar</span>
+      <a href="javascript:void(0)">Dolar</a>
     `
   }
```
Comment (anna-lach) [inline packages/components/breadcrumbs/src/breadcrumbs.stories.ts:36]: So breadcrumbs can be moved from `in progress` to `components`? :)

image

File: packages/components/breadcrumbs/src/breadcrumbs.scss [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+  --_link-gap: var(--sl-space-breadcrumb-gap-home);
+  --_link-color: var(--sl-color-href-idle-foreground);
   --_link-decoration: underline;
-  --_link-padding-block: 0.375rem;
-  --_link-padding-inline: 0.5rem;
+  --_link-padding-block: calc(var(--sl-space-breadcrumb-gap-full) / 2);
+  --_link-padding-inline: calc(var(--sl-space-breadcrumb-gap-full) / 2);
   --_link-underline-offset: 2px;
   --_tooltip-max-inline-size: min(18rem, 50vw);
+  --_divider-color: var(--sl-color-breadcrumb-divider) align-items: center;
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.scss:12]: Formatting error?

File: packages/components/breadcrumbs/src/breadcrumbs.scss [hunk 4]
```diff
@@ -43,7 +60,26 @@ a {
...
+  --_link-color: var(--sl-color-href-active-foreground);
+  font: var(--sl-text-breadcrumb-current);
+  text-decoration: none;
+}
+
+sl-popover a {
+  display: flex;
+}
+
+sl-icon[name='slash-forward'] {
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.scss:74]: Should the divider be themable?

File: packages/components/breadcrumbs/src/breadcrumbs.spec.ts [hunk 5]
```diff
@@ -146,12 +145,12 @@ describe('sl-breadcrumbs', () => {
       expect(children[7]).to.be.displayed; // <sl-icon>
       expect(children[8]).to.be.displayed; // <a>5</a>
       expect(children[9]).to.be.displayed; // <sl-icon>
-      expect(children[10]).to.be.displayed; // <span>6</span>
+      expect(children[10]).to.be.displayed; // <a>6</a>
     });
 
     it('should have a menu button to show the rest of the breadcrumbs', () => {
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.spec.ts:151]: It's not a menu button anymore, right? Just a regular button with a popover?

File: packages/components/breadcrumbs/src/breadcrumbs.spec.ts [hunk 6]
```diff
@@ -217,12 +204,12 @@ describe('sl-breadcrumbs', () => {
...
       expect(children[8]).to.be.displayed; // <a>5</a>
       expect(children[9]).to.be.displayed; // <sl-icon>
-      expect(children[10]).to.be.displayed; // <span>6</span>
+      expect(children[10]).to.be.displayed; // <a>6</a>
     });
 
     it('should show all hidden links in the menu', () => {
-      const menuButton = el.renderRoot.querySelector('sl-menu-button'),
-        menuItems = Array.from(menuButton?.querySelectorAll('sl-menu-item') ?? []);
+      const menuButton = el.renderRoot.querySelector('sl-button'),
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.spec.ts:211]: Same here, not a menu button anymore.

File: packages/components/breadcrumbs/src/breadcrumbs.spec.ts [hunk 7]
```diff
@@ -217,12 +204,12 @@ describe('sl-breadcrumbs', () => {
       expect(children[7]).to.be.displayed; // <sl-icon>
       expect(children[8]).to.be.displayed; // <a>5</a>
       expect(children[9]).to.be.displayed; // <sl-icon>
-      expect(children[10]).to.be.displayed; // <span>6</span>
+      expect(children[10]).to.be.displayed; // <a>6</a>
     });
 
     it('should show all hidden links in the menu', () => {
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.spec.ts:210]: Not a menu anymore :)

File: packages/components/breadcrumbs/src/breadcrumbs.ts [hunk 8]
```diff
@@ -67,9 +68,9 @@ export class Breadcrumbs extends ScopedElementsMixin(LitElement) {
   static get scopedElements(): ScopedElementsMap {
     return {
       'sl-icon': Icon,
-      'sl-menu-button': MenuButton,
-      'sl-menu-item': MenuItem,
-      'sl-tooltip': Tooltip
+      'sl-tooltip': Tooltip,
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.ts:71]: Please sort alphabetically

File: packages/components/breadcrumbs/src/breadcrumbs.ts [hunk 9]
```diff
@@ -126,30 +127,37 @@ export class Breadcrumbs extends ScopedElementsMixin(LitElement) {
...
+            <sl-icon name="slash-forward"></sl-icon>
           `}
       ${this.breadcrumbs.length > this.collapseThreshold
         ? html`
-            <sl-menu-button fill="link">
-              <sl-icon name="ellipsis" slot="button"></sl-icon>
+            <sl-button fill="link" id="button" @click=${this.#onClick} icon-only aria-label=${msg('More breadcrumbs')}>
+              <sl-icon name="ellipsis"></sl-icon>
+            </sl-button>
+            <sl-popover anchor="button">
```
Comment (anna-lach) [inline packages/components/breadcrumbs/src/breadcrumbs.ts:138]: There are some inconsistencies with Figma in the overflow version?

https://www.figma.com/file/1oIVcD9VMbOtLXYp48ehsA/Sanoma-Learning-Light-mode?type=design&node-id=6007-29995&mode=dev


Figma:
image


Storybook:
image

File: packages/components/breadcrumbs/src/breadcrumbs.ts [hunk 10]
```diff
@@ -126,30 +127,37 @@ export class Breadcrumbs extends ScopedElementsMixin(LitElement) {
...
-            <a href=${this.homeUrl}><sl-icon name="house"></sl-icon>${isMobile() ? '' : msg('Home')}</a>
-            <sl-icon name="chevron-right"></sl-icon>
+            <a href=${this.homeUrl}><sl-icon name="home-blank"></sl-icon>${isMobile() ? '' : msg('Home')}</a>
+            <sl-icon name="slash-forward"></sl-icon>
           `}
       ${this.breadcrumbs.length > this.collapseThreshold
         ? html`
-            <sl-menu-button fill="link">
-              <sl-icon name="ellipsis" slot="button"></sl-icon>
+            <sl-button fill="link" id="button" @click=${this.#onClick} icon-only aria-label=${msg('More breadcrumbs')}>
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.ts:135]: `icon-only` is an internal thing. I don't think we should set that here. Also not necessary?

File: packages/components/breadcrumbs/src/breadcrumbs.ts [hunk 11]
```diff
@@ -126,30 +127,37 @@ export class Breadcrumbs extends ScopedElementsMixin(LitElement) {
...
+                .map(({ url, label }) => (url ? html`<a href=${url}>${label}</a>` : html`${label}`))}
+            </sl-popover>
           `
         : nothing}
       <slot @slotchange=${this.#onSlotchange}></slot>
     `;
   }
 
+  #onClick = (): void => {
+    this.renderRoot.querySelector<Popover>('sl-popover')?.togglePopover();
```
Comment (jpzwarte) [inline packages/components/breadcrumbs/src/breadcrumbs.ts:150]: The casting top `<Popover>` should not be necessary anymore with the new `HTMLElementTagNameMap` at the top of `popover.ts`.

File: packages/locales/src/nl.ts [hunk 12]
```diff
@@ -29,5 +29,6 @@ export const templates = {
   sd0b6082239185272: 'Selecteer alles',
   sd244b51f6d7de8e2: 'Voer een waarde in volgens het gevraagde patroon.',
   sde6884478b7cc818: str`Voer tenminste ${0} karakters in (je hebt op dit momoment ${1} karakter${2}).`,
-  sf1ec4acb8d744ed9: 'Mededeling'
+  sf1ec4acb8d744ed9: 'Mededeling',
+  sf677da98fa27f9b6: 'More breadcrumbs'
```
Comment (jpzwarte) [inline packages/locales/src/nl.ts:33]: Missing translation

**Review by jpzwarte — APPROVED** [review]

File: packages/components/breadcrumbs/src/breadcrumbs.ts (hunk 10 above)
Comment (Diaan) [reply packages/components/breadcrumbs/src/breadcrumbs.ts:135]: 🙈 apparently not all Max component behaviours are out of my system... We needed to add this manually :D

File: packages/components/breadcrumbs/src/breadcrumbs.ts (hunk 9 above)
Comment (jpzwarte) [reply packages/components/breadcrumbs/src/breadcrumbs.ts:138]: I think this is a case where Figma needs to be updated with Storybook :)

File: packages/components/breadcrumbs/src/breadcrumbs.stories.ts (hunk 2 above)
Comment (Diaan) [reply packages/components/breadcrumbs/src/breadcrumbs.stories.ts:36]: I think we only should do this once we also add them to the documentation website. If teams are really eager to start using it we can let them, but if we wait to move it to components until the documentation is ready we might save our selves some work answering questions.

**Review by anna-lach — APPROVED** [review]

**Pull request merged**